src/components/base/outline-alert/outline-alert.ts

```diff
@@ -0,0 +1,78 @@
+import { html, TemplateResult, CSSResultGroup } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import componentStyles from './outline-alert.css.lit';
+import { OutlineElement } from '../outline-element/outline-element';
+
+export const alertSizes = ['small', 'large'] as const;
+export type AlertSize = typeof alertSizes[number];
+
+export const alertStatusTypes = [
+  'information',
+  'warning',
+  'error',
+  'success',
+] as const;
+export type AlertStatusType = typeof alertStatusTypes[number];
+
+// This can be useful for testing.
+export interface OutlineAlertInterface extends HTMLElement {
+  statusType: AlertStatusType;
+  size: AlertSize;
+  isInteractive: Boolean;
+  shouldShowIcon: Boolean;
+}
+
+/**
+ * The Outline Alert component
+ *
+ * @element outline-alert
+ * @slot default - The alert contents
+ * @slot outline-alert--header - The header in the alert
+ */
+@customElement('outline-alert')
+export class OutlineAlert
+  extends OutlineElement
+  implements OutlineAlertInterface
+{
+  static styles: CSSResultGroup = [componentStyles];
+
+  @property({ type: String })
+  statusType: AlertStatusType = 'information';
+
+  /**
+   * This is important context for screen readers.
+   */
+  @property({ type: Boolean })
+  isInteractive = false;
+
+  @property({ type: Boolean })
+  shouldShowIcon = true;
+
+  @property({ type: String })
+  size: AlertSize = 'large';
+
+  render(): TemplateResult {
+    // The `body` wrapper is used to avoid styles (like border) that are preventing us from styling `:host`.
+    return html`
+      <div id="body" role="${this.isInteractive ? 'alertdialog' : 'alert'}">
+        ${this.shouldShowIcon === true
```

Comment:
@damontgomery This is looking pretty good! One piece I really would like to see changed are these two conditionals placed in their own template logic. It gets a little messy having these all in the render function as is

My suggestion is to make a couple template functions below the render function for each conditional:

```
  iconTemplate = (): TemplateResult | undefined => {
    if (this.shouldShowIcon) return

    return html`
      <div id="icon"></div>
    `
  }

  largeTemplate = (): TemplateResult | undefined => {
    if (this.size !== 'large') return

    return html`
      <div id="header">
        ${this.statusType}
        <slot name="outline-alert--header"></slot>
      </div>
    `
  }
```

then the render function can just call those and it makes it much easier to reason about imo:
```
  render(): TemplateResult {
    // The `body` wrapper is used to avoid styles (like border) that are preventing us from styling `:host`.
    return html`
      <div id="body" role="${this.isInteractive ? 'alertdialog' : 'alert'}">
        ${this.iconTemplate()}
        ${this.largeTemplate()}
        <div id="message">
          <slot></slot>
        </div>
      </div>
    `;
  }

```

I think we can also lose the comment on line 55, I don't think it's of benefit.